### PR TITLE
feat(cli): cancel protection and increase polling

### PR DESCRIPTION
**Pull request by vitorveiga**

### CODE REVIEW

File: packages/jscrambler-cli/src/index.js [hunk 1]
```diff
@@ -407,12 +410,26 @@ export default {
...
+      .once('SIGTERM', onExitCancelProtection.bind(this));
+
     const protection = await this.pollProtection(
       client,
       applicationId,
       protectionId,
       await getProtectionDefaultFragments(client)
     );
+
+    process.off('SIGINT', onExitCancelProtection).off('SIGTERM', onExitCancelProtection);
```
Comment (brenopacheco-jscrambler) [inline packages/jscrambler-cli/src/index.js:431]: This off/removeListener is not removing anything

File: packages/jscrambler-cli/src/index.js [hunk 2]
```diff
@@ -567,10 +584,23 @@ export default {
...
+
+    process.once('SIGINT', onExitCancelInstrumentation.bind(this))
+      .once('SIGTERM', onExitCancelInstrumentation.bind(this));
+
     instrumentation = await this.pollInstrumentation(
       client,
       instrumentation.data.id
     );
+
+    process.off('SIGINT', onExitCancelInstrumentation).off('SIGTERM', onExitCancelInstrumentation);
```
Comment (brenopacheco-jscrambler) [inline packages/jscrambler-cli/src/index.js:602]: Same here. This statement does not remove any listener

File: packages/jscrambler-cli/src/index.js [hunk 3]
```diff
@@ -22,6 +22,9 @@ const {intoObjectType} = introspection;
 
 const debug = !!process.env.DEBUG;
 const APP_URL = 'https://app.jscrambler.com';
+const POLLING_MIN_INTERVAL = 1000;
+// in seconds
+const INCREASE_POLL_INTERVAL_EVERY = 60000
```
Comment (brenopacheco-jscrambler) [inline packages/jscrambler-cli/src/index.js:27]: This works. Is a minute too much?

File: packages/jscrambler-cli/src/index.js [hunk 4]
```diff
@@ -813,7 +844,7 @@ export default {
         case 'FINISHED_INSTRUMENTATION':
           return instrumentation;
         default:
-          await new Promise(resolve => setTimeout(resolve, 500));
+          await new Promise(resolve => setTimeout(resolve, POLLING_MIN_INTERVAL * Math.ceil((Date.now() - start) / INCREASE_POLL_INTERVAL_EVERY)));
```
Comment (brenopacheco-jscrambler) [inline packages/jscrambler-cli/src/index.js:847]: Maybe refactor this interval into a function?

File: packages/jscrambler-cli/src/index.js (hunk 1 above)
Comment (vitorveiga) [reply packages/jscrambler-cli/src/index.js:443]: done!

File: packages/jscrambler-cli/src/index.js (hunk 2 above)
Comment (vitorveiga) [reply packages/jscrambler-cli/src/index.js:614]: Done

File: packages/jscrambler-cli/src/index.js (hunk 3 above)
Comment (vitorveiga) [reply packages/jscrambler-cli/src/index.js]: i guess we can reduce it to 30000 ms and add an upper limit of 10 seconds

File: packages/jscrambler-cli/src/index.js (hunk 4 above)
Comment (vitorveiga) [reply packages/jscrambler-cli/src/index.js]: done

**Pull request merged**